### ISSUE TITLE
Relax numpy API requirement on wheel build

### ISSUE THREAD

**Issue by mhasself** [simonsobs/so3g#143]
It looks like so3g wheels are currently built against some "latest" numpy available.  If that numpy version API is higher than the one on the end system, the compiled wheels will warn, like:
```
RuntimeError: module compiled against API version 0x10 but this version of numpy is 0xe . Check the section C-API incompatibility at the Troubleshooting ImportError section at https://numpy.org/devdocs/user/troubleshooting-importerror.html#c-api-incompatibility for indications on how to solve this problem .
```
I am getting this on python3.8 with numpy 1.21.  That numpy was released less than 2 years ago and there's no reason we couldn't be supporting it.

A solution to this is to build the wheels against a pinned version of numpy that exposes a minimum supported numpy API version.  This is done in pixell for example, see how numpyver is applied in the [workflow](https://github.com/simonsobs/pixell/blob/2e51f672f9e2ea7e31439c1de3926baadfdb1621/.github/workflows/build.yml).

Pinning a numpy version in this way does not prevent the end user from using newer versions of numpy and taking advantage of bug fixes etc ... because newer numpy supports all earlier APIs.

I can take a stab at this but I know it's @tskisner favorite thing too ...

**Comment by tskisner** [comment]
Thanks for the prod that I need to finish the in-progress wheel updates.  The current numpy build lines are:
https://github.com/simonsobs/so3g/blob/909d758ce96a09c7b9f51f542a98dd5627ce276d/wheels/install_deps_linux.sh#L32-L46
In my PR in progress I added a line for 3.11 set to minimum numpy 1.24.  So the environment where you are seeing this is a conda env with python-3.8 and numpy-1.21?  I can try to replicate that while testing the new wheel.

**Comment by mhasself** [comment]
Right, I just found that ... the pinning code looks good, it's just not working...

Yes, conda env with python 3.8 and numpy 1.21.

If you look at a recent build log (e.g. here or here), it shows that numpy is installed twice ... once at 1.19 (which is our request) and again at 1.24 (which is due to some other dependency requiring >=1.20 apparently).

Incidentally, the linux_deps script installs numpy<$numpy_ver but should probably be numpy<=$numpy_ver.

**Comment by tskisner** [comment]
For reference, [here are the Numpy ABI versions](https://github.com/numpy/numpy/blob/e42c9503a14d66adfd41356ef5640c6975c45218/numpy/core/include/numpy/numpyconfig.h#L65-L83).

Note that the `<` sign is intentional (i.e. we want a version less that 1.20, which is the version where the ABI changed).  I verified the problem locally in a conda env with python-3.8.  The latest available astropy for py38 (5.2.1) requires numpy>=1.20 (so they are building against the newer ABI).  I fixed this by adding our numpy restriction to the same pip line where our requirements file is parsed.  So, our choices are:

1.  Continue to force numpy < 1.20 (i.e. 1.19 on freshly installed environments), and limit ourselves to the older astropy-5.1.1
2.  Build against the newer ABI (numpy 1.20), supporting envs with astropy-5.2, but possibly breaking existing py38 envs that have older versions of numpy and where the user is pip-installing so3g and does not want to upgrade other packages.

For newly created environments, either is fine (and likely the user would be creating that with python-3.10 anyway).  For pip installing so3g into old, existing python 3.8 environments, we probably want option (1).  Happy to implement either option.
